data-prepper-plugins/kafka-plugins/src/main/java/org/opensearch/dataprepper/plugins/kafka/consumer/KafkaCustomConsumerFactory.java

```diff
@@ -0,0 +1,243 @@
+                    break;
+                case CANONICAL_BOOTSTRAP:
+                    properties.put("client.dns.lookup", ClientDNSLookupType.CANONICAL_BOOTSTRAP.toString());
+                    break;
+                case DEFAULT:
+                    properties.put("client.dns.lookup", ClientDNSLookupType.DEFAULT.toString());
+                    break;
+            }
+        }
+        setConsumerTopicProperties(properties, topicConfig);
+        setSchemaRegistryProperties(sourceConfig, properties, topicConfig);
+        LOG.info("Starting consumer with the properties : {}", properties);
+        return properties;
+    }
+
+    private void setConsumerTopicProperties(final Properties properties, final TopicConfig topicConfig) {
+        properties.put(ConsumerConfig.GROUP_ID_CONFIG, topicConfig.getGroupId());
+        properties.put(ConsumerConfig.MAX_PARTITION_FETCH_BYTES_CONFIG, (int)topicConfig.getMaxPartitionFetchBytes());
+        properties.put(ConsumerConfig.RETRY_BACKOFF_MS_CONFIG, ((Long)topicConfig.getRetryBackoff().toMillis()).intValue());
+        properties.put(ConsumerConfig.RECONNECT_BACKOFF_MS_CONFIG, ((Long)topicConfig.getReconnectBackoff().toMillis()).intValue());
+        properties.put(ConsumerConfig.ENABLE_AUTO_COMMIT_CONFIG,
+            topicConfig.getAutoCommit());
+        properties.put(ConsumerConfig.AUTO_COMMIT_INTERVAL_MS_CONFIG,
+            ((Long)topicConfig.getCommitInterval().toMillis()).intValue());
+        properties.put(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG,
+            topicConfig.getAutoOffsetReset());
+        properties.put(ConsumerConfig.MAX_POLL_RECORDS_CONFIG,
+            topicConfig.getConsumerMaxPollRecords());
+        properties.put(ConsumerConfig.MAX_POLL_INTERVAL_MS_CONFIG,
+            ((Long)topicConfig.getMaxPollInterval().toMillis()).intValue());
+        properties.put(ConsumerConfig.SESSION_TIMEOUT_MS_CONFIG, ((Long)topicConfig.getSessionTimeOut().toMillis()).intValue());
+        properties.put(ConsumerConfig.HEARTBEAT_INTERVAL_MS_CONFIG, ((Long)topicConfig.getHeartBeatInterval().toMillis()).intValue());
+        properties.put(ConsumerConfig.FETCH_MAX_BYTES_CONFIG, (int)topicConfig.getFetchMaxBytes());
+        properties.put(ConsumerConfig.FETCH_MAX_WAIT_MS_CONFIG, topicConfig.getFetchMaxWait());
+        properties.put(ConsumerConfig.FETCH_MIN_BYTES_CONFIG, (int)topicConfig.getFetchMinBytes());
+    }
+
+    private void setSchemaRegistryProperties(final KafkaConsumerConfig kafkaConsumerConfig, final Properties properties, final TopicConfig topicConfig) {
+        SchemaConfig schemaConfig = kafkaConsumerConfig.getSchemaConfig();
+        if (Objects.isNull(schemaConfig)) {
+            setPropertiesForPlaintextAndJsonWithoutSchemaRegistry(properties, topicConfig);
+            return;
+        }
+
+        if (schemaConfig.getType() == SchemaRegistryType.AWS_GLUE) {
+            return;
+        }
+
+        /* else schema registry type is Confluent */
```

Comment:
Comments like these get lost over time. It would be better to move this into a new method. This keeps the relevant code together and gives a natural name to what is happening.

```
setupConfluentSchemaRegistry(...);
```